src/access/ownable/mixins.cairo

```diff
@@ -1,35 +1,80 @@
+use starknet::ContractAddress;
+
+#[starknet::interface]
+trait IOwnableMixin<TState> {
+    // IOwnable
+    fn owner(self: @TState) -> ContractAddress;
+    fn transfer_ownership(ref self: TState, new_owner: ContractAddress);
+    fn renounce_ownership(ref self: TState);
+
+    // IOwnableCamelOnly
+    fn transferOwnership(ref self: TState, newOwner: ContractAddress);
+    fn renounceOwnership(ref self: TState);
+}
+
 #[starknet::component]
-mod OwnableDual {
-    use openzeppelin::access::ownable::interface;
+mod OwnableMixin {
+    use openzeppelin::access::ownable::OwnableComponent::{OwnableImpl, OwnableCamelOnlyImpl};
+    use openzeppelin::access::ownable::OwnableComponent;
     use starknet::ContractAddress;
 
     #[storage]
     struct Storage {}
 
-    #[embeddable_as(OwnableDualImpl)]
-    impl IOwnableDual<
-        TContractState, +HasComponent<TContractState>
-    > of interface::IOwnableDual<ComponentState<TContractState>> {
+    #[embeddable_as(OwnableMixinImpl)]
+    impl OwnableMixin<
+        TContractState,
+        +HasComponent<TContractState>,
+        +OwnableComponent::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of super::IOwnableMixin<ComponentState<TContractState>> {
         fn owner(self: @ComponentState<TContractState>) -> ContractAddress {
-            self.owner()
+            let ownable = self.get_ownable();
+            ownable.owner()
         }
 
         fn transfer_ownership(
             ref self: ComponentState<TContractState>, new_owner: ContractAddress
         ) {
-            self.transfer_ownership(new_owner);
+            let mut ownable = self.get_ownable_mut();
+            ownable.transfer_ownership(new_owner);
         }
 
         fn renounce_ownership(ref self: ComponentState<TContractState>) {
-            self.renounce_ownership();
+            let mut ownable = self.get_ownable_mut();
+            ownable.renounce_ownership();
         }
 
         fn transferOwnership(ref self: ComponentState<TContractState>, newOwner: ContractAddress) {
-            self.transferOwnership(newOwner);
+            let mut ownable = self.get_ownable_mut();
+            ownable.transferOwnership(newOwner);
         }
 
         fn renounceOwnership(ref self: ComponentState<TContractState>) {
-            self.renounceOwnership();
+            let mut ownable = self.get_ownable_mut();
+            ownable.renounceOwnership();
+        }
+    }
+
+    #[generate_trait]
+    impl GetOwnableImpl<
```

Comment:
Correct, we don't need them for `get_dep_component_mut`, but we can still use them for view functions because snapshots don't seem to be supported yet with `get_dep_component`

Reply:
they're now right?

Reply:
> they're now right?

They are, indeed